src/utils.rs

```diff
@@ -15,3 +18,18 @@ where
 {
     empty::<T>().collect()
 }
+
+/// Union two BTreeMaps, call f to resolve conflicts if duplicate keys are encountered.
+pub fn union_b_tree_maps_with<K: Clone + Ord, V: Clone, F: Fn(&V, &V) -> V>(
+    f: F,
+    l: &BTreeMap<K, V>,
+    r: &BTreeMap<K, V>,
```

Comment:
Just a thought: I think it would also be okay to remove the borrows here, and let the function user decide, if they want to consume or clone the original maps.
```suggestion
    l: BTreeMap<K, V>,
    r: BTreeMap<K, V>,
```

Reply:
Had to modify the impl a bit as `entry` doesn't hand over the ownership of the lhs.